### ISSUE TITLE
Undefined method 'finalize!' after installing 1.6.5

### ISSUE THREAD

**Issue by chilimangoes** [mitchellh/vagrant#4714]
I just upgraded to Vagrant 1.6.5 from 1.4.x and I now get the following error calling any time I call vagrant. This exact error seems to happen with all vagrant commands, including just calling status on all of my vagrant projects.

    C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.6.5/lib/vagrant/config/v2/root.rb:50:in `block in finalize!': undefined method `finalize!' for []:Array (NoMethodError)

Full stack trace and debug log here:
https://gist.github.com/anonymous/98c5bc64d775aa81ce53

**Comment by sethvargo** [comment]
@chilimangoes is this still an issue with Vagrant 1.7.1? I cannot reproduce the issue with your Vagrantfile. It is also possible that you have an old/outdated plugin that is causing the problem.

**Comment by chilimangoes** [comment]
Hi @sethvargo, thanks for the plugin suggestion. Turns out it was an old version of the vagrant-mountcommand plugin (which is no longer needed as of Vagrant 1.6). I uninstalled that plugin and Vagrant is working again.

It seems like this should be considered a bug in Vagrant. I'm not familiar with the Vagrant plugin architecture, but would it be possible to wrap the loading and execution of plugins in some error handling and then print out a more useful error to the user? The stack trace and debug logs in my original post don't seem to provide any clue that the problem was plugin related.
